dev-tools/oxlog/src/bin/oxlog.rs

```diff
@@ -0,0 +1,116 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Tool for discovering oxide related logfiles on sleds
+
+use clap::{Parser, Subcommand};
+use oxlog::{LogFile, Zones};
+
+#[derive(Debug, Parser)]
+#[command(version)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Debug, Subcommand)]
+enum Commands {
+    /// List all zones found on the filesystem
+    Zones,
+
+    /// List logs for a given service
+    Logs {
+        // The name of the zone
+        zone: String,
+
+        /// The name of the service to list logs for
+        service: Option<String>,
+
+        /// Print available metadata
+        #[arg(short, long)]
+        metadata: bool,
+
+        /// Print only the current log file
+        #[arg(short, long)]
+        current: bool,
+
+        /// Print only the archive log files
+        #[arg(short, long)]
+        archived: bool,
```

Comment:
I think there is a third class, rotated log files that have not yet been archived. These probably don't live very long, since the archiver runs frequently. But we may want to consider including them and a CLI switch to filter them.

Reply:
These are actually included with the archived files right now. I"m happy to separate them out though if you think that makes sense.